File: consultoria/src/main/java/br/com/sigo/consultoria/enums/EnumAtributosPesquisaUsuario.java
package br.com.sigo.consultoria.enums;

import lombok.AllArgsConstructor;

import java.io.Serializable;

@AllArgsConstructor
public enum EnumAtributosPesquisaUsuario implements Serializable {

  ID("id", "id", ""),
  NOME("nome", "nome", ""),
  EMPRESA("empresa", "empresa_id", "usuario_id"),
  ;

  private String param;
  private String atributo;
  private String atributoJoin;

  public String getParam() {
    return param;
  }

  public String getAtributo() {
    return atributo;
  }

  public String getAtributoJoin() {
    return atributoJoin;
  }
}

File: consultoria-app/app/src/main/java/br/com/sigo/consultoria/enums/EnumAtributosPesquisaUsuario.kt
package br.com.sigo.consultoria.enums

import java.io.Serializable

enum class EnumAtributosPesquisaUsuario(
    val param: String,
    val atributo: String,
    val atributoJoin: String
) : Serializable {
    ID("id", "id", ""), NOME("nome", "nome", ""), EMPRESA("empresa", "empresa_id", "usuario_id");

}